mitodl/open-discussions
# add user comment deletion

#### What are the relevant tickets?

closes #233 

#### What's this PR do?

this adds deletion functionality for the user's own comments.

#### How should this be manually tested?

Basically, if you're looking at your own comments you should see a 'delete' button. Clicking the button should open a confirmation dialog. If you click 'yes' the comment should update to look like this in the UI:

![fjfjfjasdfasdfasdf](https://user-images.githubusercontent.com/6207644/34117525-f796a1e6-e3e9-11e7-8255-9eeef1956f59.png)

If you refresh the page you should see one of two things: if the comment you deleted did not have any replies it should just disappear, and if the comment in question had at least one reply you should see something like the above picture, with the replies sticking around under a 'deleted' placeholder.

If you're looking at someone else's comments you shouldn't see the 'delete' button.
